payu/models/um.py

```diff
@@ -109,19 +110,29 @@ def collate(self):
         pass
 
     def setup(self):
+        # Raise a depreciation error if the um_env.yaml file is not found
+        # This could be removed down the line, once older configurations 
+        # have swapped to um_env.yaml files.
+        depreciated_um_env = os.path.join(self.control_path, 'um_env.py')
+        new_um_env = os.path.join(self.control_path, 'um_env.yaml')
+        if (not os.path.isfile(new_um_env)) and os.path.isfile(depreciated_um_env):
+            raise FutureWarning(
+                (
+                    "The `um_env.py` configuration file has been depreciated and "
+                    "should be relplaced with a yaml file. "
+                    "Convert `um_env.py` to `um_env.yaml` using "
+                    "https://github.com/ACCESS-NRI/esm1.5-scripts/blob/main/config-files/UM/um_env_to_yaml.py"
+                )
+            ) 
+
+        # Commence normal setup
         super(UnifiedModel, self).setup()
 
         # Set up environment variables needed to run UM.
-        # Look for a python file in the config directory.
-        loader = SourceFileLoader(
-            'um_env',
-            os.path.join(self.control_path, 'um_env.py')
-        )
-        um_env = module_from_spec(
-            spec_from_loader(loader.name, loader)
-        )
-        loader.exec_module(um_env)
-        um_vars = um_env.vars
+        um_env_path = os.path.join(self.control_path, 'um_env.yaml')
```

Comment:
Just to be sure, there's no possibility of someone having their file name be the short form (`um_env.yml`)?

Reply:
I think it should be unlikely, as the released configs will have a `um_env.yaml` file, and if the conversion script is used for an older configuration it will also default to a `yaml` suffix, unless the user specifies an output path with a shorter `.yml` suffix.

Or, at least I hope so! Right now it looks like payu isn't able to handle alternative names for any of the compulsory configuration files.